src/main/java/pw/itr0/kaba/encrypt/PasswordBasedEncrypter.java

```diff
@@ -30,9 +52,26 @@
      * @param password encrypting password
      * @param secret   secret bytes
      * @return encrypted secret (Base64 encoded string)
-     * @throws GeneralSecurityException on encryption failure
+     * @throws InvalidKeyException       if the {@code password} does not satisfy password specification of using encryption algorithm,
+     *                                   or encryption algorithm using specified password length is not usable on current JDK
+     * @throws BadPaddingException       if the {@code secret} is not properly padded
+     * @throws IllegalBlockSizeException if the {@code secret} has wrong length
      */
-    default String encrypt64(char[] password, byte[] secret) throws GeneralSecurityException {
+    default String encrypt64(char[] password, byte[] secret)
+            throws InvalidKeyException, BadPaddingException, IllegalBlockSizeException {
         return Base64.getEncoder().encodeToString(this.encrypt(password, secret));
     }
+
+    /**
+     * Wrap {@link OutputStream} {@code out} to {@link CipherOutputStream} encrypting with {@code password}.
+     *
+     * @param out      {@link OutputStream} to wrap in {@link CipherOutputStream}
+     * @param password encrypting password
+     * @return encrypting {@link OutputStream}
+     * @throws InvalidKeyException if the {@code password} does not satisfy password specification of using encryption algorithm,
+     *                             or encryption algorithm using specified password length is not usable on current JDK
+     */
+    default OutputStream wrap(OutputStream out, char[] password) throws InvalidKeyException {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid prefixing parameters by in, out or inOut. Uses Javadoc to document this behavior.](https://www.codacy.com/app/ryotan/kaba-vault/file/2671050500/issues/source?bid=3298589&fileBranchId=3332567#l74)